app/controllers/spree/products_controller_decorator.rb

```diff
@@ -1,9 +1,18 @@
 module Spree
   ProductsController.class_eval do
-    after_action :recently_viewed, only: :show
+    skip_before_action :set_current_order, only: :recently_viewed
+    after_action :save_recently_viewed, only: :recently_viewed
```

Comment:
@priyank-gupta if show action will be cached it won't fire, that's why it's after recently_viewed